enhancements/network/user-defined-network-segmentation.md

```diff
@@ -0,0 +1,966 @@
+---
+title: user-defined-network-segmentation
+authors:
+  - "@trozet"
+  - "@qinqon"
+reviewers:
+  - "@tssurya"
+  - "@danwinship"
+  - "@fedepaol"
+  - "@maiqueb"
+  - "@jcaamano"
+  - "@Miciah"
+  - "@dceara"
+  - "@dougbtv"
+approvers:
+  - "@tssurya"
+  - "@jcaamano"
+api-approvers:
+  - "None"
+creation-date: 2024-05-03
+last-updated: 2024-05-28
+tracking-link:
+  - https://issues.redhat.com/browse/SDN-4789
+---
+
+# User-Defined Network Segmentation
+
+## Summary
+
+OVN-Kubernetes today allows multiple different types of networks per secondary network: layer 2, layer 3, or localnet.
+Pods can be connected to different networks without discretion. For the primary network, OVN-Kubernetes only supports all
+pods connecting to the same layer 3 virtual topology. The scope of this effort is to bring the same flexibility of the
+secondary network to the primary network. Therefore, pods are able to connect to different types of networks as their
+primary network.
+
+Additionally, multiple and different instances of primary networks may co-exist for different users, and they will provide
+native network isolation.
+
+## Terminology
+
+* **Primary Network** - The network which is used as the default gateway for the pod. Typically recognized as the eth0
+interface in the pod.
```

Comment:
This is going be confusing for Dual Stack as well.  We should always have two default gateways, one for each protocol and they should not necessarily be confined to the same interface.

I guess the real "default gateway" will be chosen by Happy Eyeballs and whichever protocol is faster.

Reply:
Today we always map the primary interface as the default gateway for both protocols, but its a good point that in the future someone may want a primary network for v4 and a different primary network for v6. @tssurya @ormergi we should keep this in mind as we discuss what the API should be for creating networks, and make sure we dont constrain ourselves from expanding to this in the future